src/components/mask-input/mask-input-base.ts

```diff
@@ -0,0 +1,153 @@
+import { property, state } from 'lit/decorators.js';
+import { IgcInputBaseComponent } from '../input/input-base';
+import { MaskParser } from './mask-parser';
+interface MaskSelection {
+  start: number;
+  end: number;
+}
+
+export abstract class IgcMaskInputBaseComponent extends IgcInputBaseComponent {
+  protected parser = new MaskParser();
+  protected selection: MaskSelection = { start: 0, end: 0 };
+  protected compositionStart = 0;
+
+  @state()
+  protected hasFocus = false;
+
+  @state()
+  protected maskedValue = '';
+
+  /** The mask pattern to apply on the input. */
+  @property()
+  public mask!: string;
+
+  /** The prompt symbol to use for unfilled parts of the mask. */
+  @property()
+  public prompt!: string;
+
+  /** The direction attribute of the control. */
+  @property({ reflect: true })
+  public override dir: 'ltr' | 'rtl' | 'auto' = 'auto';
```

Comment:
There's a type `Direction` exposed on the input - not sure we care to reuse it. Or if we should be reusing this across all components even. Might be a useful indicator for further generated API to reuse the same type as well.

Also if every type of input will be overriding this separately, perhaps we should just move the dir to the input base?
// CC @rkaraivanov